istioctl/pkg/install/verify.go

```diff
@@ -0,0 +1,212 @@
+				Get().
+				Resource(kinds).
+				Namespace(namespace).
+				Name(name).
+				VersionedParams(&meta_v1.GetOptions{}, scheme.ParameterCodec).
+				Do().
+				Into(deployment)
+			if err != nil {
+				return err
+			}
+			err = getDeploymentStatus(deployment, name)
+			if err != nil {
+				return err
+			}
+		case "Job":
+			job := &v1batch.Job{}
+			err = info.Client.
+				Get().
+				Resource(kinds).
+				Namespace(namespace).
+				Name(name).
+				VersionedParams(&meta_v1.GetOptions{}, scheme.ParameterCodec).
+				Do().
+				Into(job)
+			if err != nil {
+				return err
+			}
+			for _, c := range job.Status.Conditions {
+				switch c.Type {
+				case v1batch.JobFailed:
+					return fmt.Errorf("istio installation fails - the required Job  %s failed", name)
+				}
+			}
+		default:
+			result := info.Client.
+				Get().
+				Resource(kinds).
+				Name(name).
+				Do()
+			if result.Error() != nil {
+				result = info.Client.
+					Get().
+					Resource(kinds).
+					Namespace(namespace).
+					Name(name).
+					Do()
+				if result.Error() != nil {
+					return fmt.Errorf("istio installation fails or have not been completed - the required %s:%s is not ready due to: %v", kind, name, result.Error())
+				}
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(writer, "istio is installed successfully\n")
+	return nil
+}
+
+// NewVerifyCommand creates a new command for verifying Istio Installation Status
+func NewVerifyCommand() *cobra.Command {
+	var (
+		kubeConfigFlags = &genericclioptions.ConfigFlags{
+			Context:    strPtr(""),
+			Namespace:  strPtr(""),
+			KubeConfig: strPtr(""),
+		}
+
+		filenames     = []string{}
+		fileNameFlags = &genericclioptions.FileNameFlags{
+			Filenames: &filenames,
+			Recursive: boolPtr(true),
+		}
+	)
+	verifyInstallCmd := &cobra.Command{
+		Use:   "verify-install",
+		Short: "Verifies Istio Installation Status",
+		Long: `
+		verify-install Verifies Istio Installation Status
+`,
+		Example: `
+istioctl verify-install -f istio-demo.yaml
+`,
+		RunE: func(c *cobra.Command, _ []string) error {
+			return verifyInstall(kubeConfigFlags, fileNameFlags.ToOptions(), c.OutOrStderr())
+		},
+	}
+
+	flags := verifyInstallCmd.PersistentFlags()
+	kubeConfigFlags.AddFlags(flags)
+	fileNameFlags.AddFlags(flags)
+	return verifyInstallCmd
+}
+
+func strPtr(val string) *string {
+	return &val
+}
+
+func boolPtr(val bool) *bool {
+	return &val
+}
+
+func getDeploymentStatus(deployment *v1beta1.Deployment, name string) error {
+	cond := getDeploymentCondition(deployment.Status, v1beta1.DeploymentProgressing)
+	if cond != nil && cond.Reason == "ProgressDeadlineExceeded" {
+		return fmt.Errorf("istio installation fails or have not been completed"+
+			" - deployment %q exceeded its progress deadline", name)
+	}
+	if deployment.Spec.Replicas != nil && deployment.Status.UpdatedReplicas < *deployment.Spec.Replicas {
+		return fmt.Errorf("istio installation fails or have not been completed"+
```

Comment:
would be good to have different error msg than earlier.  I think here is # of replicas for XX deployment(s) haven't reached desired number of replicas.

Reply:
ah I see, it is different msg.